pulumi/src/index.ts

```diff
@@ -58,11 +66,34 @@ export interface ServiceConfig {
   storageSize: string
 }
 
+export interface ServiceInput {
+  coinServiceName: string
+  ports: Record<string, Port>
+  command?: Array<string>
+  args?: Array<string>
+  env?: Record<string, string>
+  dataDir?: string
+  configMapData?: Record<string, string>
+  volumeMounts?: Array<k8s.types.input.core.v1.VolumeMount>
+  readinessProbe?: k8s.types.input.core.v1.Probe
+  livenessProbe?: k8s.types.input.core.v1.Probe
+}
+
+export interface JointCoinServiceInput extends ServiceInput, ServiceConfig {}
```

Comment:
Not quite, they both have required props so we can't loosely swap them

Reply:
`interface JointCoinServiceInput extends ServiceInput, ServiceConfig {}` is the same as
```
interface ServiceInput extends ServiceConfig {
   ...serviceInputFields
}
```

Reply:
Then how are you going to compile the `ServiceInput`/`ServiceArgs`? 

```
Type '{ coinServiceName: string; ports: { 'daemon-rpc': { port: number; }; }; env: { NETWORK: string; }; }' is missing the following properties from type 'ServiceArgs': image, name, storageSizets(2739)
```
Please push this change if you see it can work, I still don't